sha3/src/main/java/com/theromus/sha/Keccak.java

```diff
@@ -16,24 +22,148 @@
+     * @throws CantSetCustomHashLen Length can be set only for SHAKE128 and SHAKE256
+     */
+    public void setHashType(final Parameters type, int customLen) throws NotValidHashLenException, CantSetCustomHashLen {
+    	this.type = type;
+    	setCustomHashLen(customLen);
+	}
+    
+    /**
+     * Check if custom length is valid. Can't be less then 0 and can be set only for SHAKE128 and SHAKE256
+     * 
+     * @param type sha type
+     * @param customLen custom length of hash
+     * @throws NotValidHashLenException Length can't be less then 0
+     * @throws CantSetCustomHashLen Length can be set only for SHAKE128 and SHAKE256
+     */
+    private static boolean isValid(Parameters type, int customLen) throws NotValidHashLenException, CantSetCustomHashLen {
+    	if(type.getOutputLen() == customLen)
+    		return true;
+    	if(customLen <= 0)
+    		throw new NotValidHashLenException();
+    	if(type != Parameters.SHAKE128 && type != Parameters.SHAKE256)
+    		throw new CantSetCustomHashLen("You can set custom output size only with SHAKE128 or SHAKE256");
+    	return true;
+    }
+    
+    /**
+     * 
+     * @param customLen custom length of hash
+     * @throws NotValidHashLenException Length can't be less then 0
+     * @throws CantSetCustomHashLen Length can be set only for SHAKE128 and SHAKE256
+     */
+    public void setCustomHashLen(int customLen) throws NotValidHashLenException, CantSetCustomHashLen {
+    	isValid(type, customLen);
+		this.customLen = customLen;
+	}
+    
+    /**
+     * Do hash.
+     * 
+     * @param message input data
+     * @return byte-array result
+     */
+    public byte[] getHash(final byte[] message) throws MustBeSetTypeHashException {
+    	if(type == null)
+    		throw new MustBeSetTypeHashException();
+		try {
+			return getHash(message, type, getOutputLen());
+		} catch (NotValidHashLenException | CantSetCustomHashLen e) {
+			e.printStackTrace();
+		}
+		return null;
+	}
+    
+    /**
+     * Do hash.
+     * 
+     * @param message input data
+     * @param type sha type
+     * @return byte-array result
+     */
+    public static byte[] getHash(final byte[] message, final Parameters type) {
+		try {
+			return getHash(message, type, type.getOutputLen());
+		} catch (NotValidHashLenException | CantSetCustomHashLen e) {
+			e.printStackTrace();
+		}
+		return null;
+	}
+    
     /**
      * Do hash.
      *
      * @param message input data
-     * @param parameter keccak param
+     * @param type sha type
+     * @param customLen custom length of hash
      * @return byte-array result
+     * @throws CantSetCustomHashLen Length can be set only for SHAKE128 and SHAKE256
+     * @throws NotValidHashLenException Length can't be less then 0
      */
-    public byte[] getHash(final byte[] message, final Parameters parameter) {
+    public static byte[] getHash(final byte[] message, final Parameters type, final int customLen) throws NotValidHashLenException, CantSetCustomHashLen {
```

Comment:
Leave only one (non-static) method with this signature com.theromus.sha.Keccak#getHash(byte[])

Reply:
I inserted the static method just to have a quick and immediate way to calculate the hash while keeping the signature and structure already present.
With the new signatures it is obviously not necessary.